src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -188,8 +188,25 @@ bool TryCreateParentDirectory(const string& prefix, const string& filename) {
 // Get the absolute path of this protoc binary.
 bool GetProtocAbsolutePath(string* path) {
 #ifdef _WIN32
-  char buffer[MAX_PATH];
-  int len = GetModuleFileName(NULL, buffer, MAX_PATH);
+  wchar_t wide_buffer[MAX_PATH];
+  // Note: the returned length does not include terminating null character.
+  int len = GetModuleFileNameW(NULL, wide_buffer, MAX_PATH);
+  if (len == GOOGLE_ARRAYSIZE(wide_buffer) &&
+      GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
+    // The path is too long.
+    return false;
+  }
+  char buffer[MAX_PATH + 1];
+  for (size_t i = 0; i < len; ++i) {
+    wchar_t wide_char = wide_buffer[i];
+    if (wide_char > 0x7F) {
+      // non-ASCII character is not supported yet.
+      // TODO(yukawa): Support paths that contain non-ASCII characters.
```

Comment:
Yes it works but its binary representation depends on the system locale. It can be [CP 932](https://en.wikipedia.org/wiki/Shift_JIS), [CP 1252](https://en.wikipedia.org/wiki/Windows-1252), or whatever Windows thinks appropriate.  Yes they are char array, but you cannot simply use `+ "/include"` against the retrieved char array because there is no guarantee that `"/include"` is a valid character sequence in the encoding used there.